src/ray/core_worker/core_worker.cc

```diff
@@ -1441,57 +1404,42 @@ static std::vector<std::string> GetUrisFromRuntimeEnv(
   return result;
 }
 
-static std::vector<std::string> GetUrisFromSerializedRuntimeEnv(
-    const std::string &serialized_runtime_env) {
+std::string CoreWorker::OverrideTaskOrActorRuntimeEnv(
+    const std::string &serialized_runtime_env,
+    std::vector<std::string> *runtime_env_uris) {
+  std::shared_ptr<rpc::RuntimeEnv> parent_runtime_env;
+  std::string parent_serialized_runtime_env;
+  if (options_.worker_type == WorkerType::DRIVER) {
+    parent_runtime_env = job_runtime_env_;
+    parent_serialized_runtime_env =
+        job_config_->runtime_env_info().serialized_runtime_env();
+  } else {
+    parent_runtime_env = worker_context_.GetCurrentRuntimeEnv();
+    parent_serialized_runtime_env = worker_context_.GetCurrentSerializedRuntimeEnv();
+  }
+  if (IsRuntimeEnvEmpty(serialized_runtime_env)) {
+    // Try to inherit runtime env from job or worker.
+    *runtime_env_uris = GetUrisFromRuntimeEnv(parent_runtime_env.get());
+    return parent_serialized_runtime_env;
+  }
+
+  if (!IsRuntimeEnvEmpty(parent_serialized_runtime_env)) {
+    RAY_LOG(WARNING) << "Runtime env already exists and the parent runtime env is "
+                     << parent_serialized_runtime_env << ". It will be overrode by "
+                     << serialized_runtime_env << ".";
```

Comment:
```suggestion
    RAY_LOG(WARNING) << "Runtime env already exists and the parent runtime env is "
                     << parent_serialized_runtime_env << ". It will be overridden by "
                     << serialized_runtime_env << ".";
```
Why is this a WARNING? Isn't it expected normal behavior?

Reply:
I add this WARNING log because current PR already changed the API behavior. It is from the advise of Edward https://github.com/ray-project/ray/issues/21818#issuecomment-1021704212. Maybe we can change this log level After a period of time? I can add a TODO here.

Reply:
Ah makes sense!  Yeah, we'll need some way of remembering to change the log level in the future.  A TODO sounds like a good solution.

Reply:
Actually, when you merge this PR could you make a quick issue for this and add it to the runtime_env backlog milestone?  I think that will help prevent it from being lost.